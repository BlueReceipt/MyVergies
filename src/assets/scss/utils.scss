.has-border-radius {
  border-radius: $box-radius;
}

.has-shadow-danger {
  box-shadow: 0 3px 30px rgba(115, 0, 0, 0.5);
}

.has-shadow-success {
  box-shadow: 0 3px 30px rgba(0, 123, 101, 0.7);
}

.has-shadow-dark {
  box-shadow: 0 3px 30px rgba(0, 68, 111, 0.5);
}

@media (prefers-color-scheme: dark) {
  .has-shadow-danger {
    box-shadow: 0 3px 30px rgba(255, 0, 36, 0.3);
  }

  .has-shadow-success {
    box-shadow: 0 3px 30px rgba(0, 255, 208, 0.3);
  }

  .has-shadow-dark {
    box-shadow: 0 3px 30px rgba(0, 156, 255, 0.3);
  }
}
