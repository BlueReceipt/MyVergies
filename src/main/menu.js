import {app, Menu} from 'electron'

/**
 * macOS Dock menu items
 */
app.dock.setMenu(Menu.buildFromTemplate([
  {
    label: 'Send',
    click () {
      console.log('Send')
    }
  }, {
    label: 'Receive',
    click () {
      console.log('Receive')
    }
  },
  {
    label: 'Settings',
    click () {
      console.log('Settings')
    }
  }
]))

/**
 * Application menu.
 */
let template = []

let actions = {
  label: 'Actions',
  submenu: [
    {
      label: 'Unlock'
    },
    {
      type: 'separator'
    },
    {
      label: 'Send',
      click () {
        console.log('Send')
      }
    },
    {
      label: 'Receive',
      click () {
        console.log('Send')
      }
    }
  ]
}

let view = {
  label: 'View',
  submenu: [
    {
      role: 'resetzoom'
    },
    {
      role: 'zoomin'
    },
    {
      role: 'zoomout'
    },
    {
      type: 'separator'
    },
    {
      role: 'togglefullscreen'
    }
  ]
}

if (process.env.NODE_ENV === 'development') {
  view.submenu.unshift({
    type: 'separator'
  })
  view.submenu.unshift({
    label: 'Toggle Developer Tools',
    accelerator: process.platform === 'darwin' ? 'Alt+Command+I' : 'Ctrl+Shift+I',
    click (item, focusedWindow) {
      if (focusedWindow) focusedWindow.webContents.toggleDevTools()
    }
  })
  view.submenu.unshift({
    label: 'Reload',
    accelerator: 'CmdOrCtrl+R',
    click (item, focusedWindow) {
      if (focusedWindow) focusedWindow.reload()
    }
  })
}

let window = {
  role: 'window',
  submenu: [
    {
      role: 'minimize'
    },
    {
      role: 'close'
    }
  ]
}

let help = {
  role: 'help',
  submenu: [
    {
      label: 'Learn how to create a new wallet'
    },
    {
      label: 'Learn how to restore an existing wallet'
    },
    {
      label: 'Learn how to send XVG'
    },
    {
      label: 'Learn how to receive XVG'
    },
    {
      type: 'separator'
    },
    {
      label: 'Visit our GitHub',
      click () {
        require('electron').shell.openExternal('https://github.com/vergecurrency')
      }
    },
    {
      label: 'Visit vergecurrency.com',
      click () {
        require('electron').shell.openExternal('https://vergecurrency.com')
      }
    }
  ]
}

template.push(actions)
template.push(view)
template.push(window)
template.push(help)

if (process.platform === 'darwin') {
  const name = app.getName()
  template.unshift({
    label: name,
    submenu: [
      {
        role: 'about'
      },
      {
        type: 'separator'
      },
      {
        label: 'Settings',
        accelerator: 'CmdOrCtrl+,'
      },
      {
        role: 'services',
        submenu: []
      },
      {
        type: 'separator'
      },
      {
        role: 'hide'
      },
      {
        role: 'hideothers'
      },
      {
        role: 'unhide'
      },
      {
        type: 'separator'
      },
      {
        role: 'quit'
      }
    ]
  })

  // Window menu.
  template[3].submenu = [
    {
      label: 'Close',
      accelerator: 'CmdOrCtrl+W',
      role: 'close'
    },
    {
      label: 'Minimize',
      accelerator: 'CmdOrCtrl+M',
      role: 'minimize'
    },
    {
      label: 'Zoom',
      role: 'zoom'
    },
    {
      type: 'separator'
    },
    {
      label: 'Bring All to Front',
      role: 'front'
    }
  ]
}

const menu = Menu.buildFromTemplate(template)
Menu.setApplicationMenu(menu)
