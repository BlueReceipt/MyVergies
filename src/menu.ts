import { app, MenuItem } from 'electron'

export interface MenuValues {
  forceQuit: boolean
}

export const menuValues: MenuValues = {
  forceQuit: false
}

/**
 * macOS Dock menu items
 */
export const dockTemplate = [
  {
    label: 'Send',
    click () {
      console.log('Send')
    }
  }, {
    label: 'Receive',
    click () {
      console.log('Receive')
    }
  },
  {
    label: 'Settings',
    click () {
      console.log('Settings')
    }
  }
]

/**
 * Application menu.
 */
export let template: MenuItem[] = []

let actions = {
  label: 'Actions',
  submenu: [
    {
      label: 'Lock wallet'
    },
    {
      type: 'separator'
    },
    {
      label: 'Send',
      click () {
        console.log('Send')
      }
    },
    {
      label: 'Receive',
      click () {
        console.log('Send')
      }
    }
  ]
}

let view = {
  label: 'View',
  submenu: [
    {
      role: 'resetzoom'
    },
    {
      role: 'zoomin'
    },
    {
      role: 'zoomout'
    },
    {
      type: 'separator'
    },
    {
      role: 'togglefullscreen'
    }
  ]
}

if (process.env.NODE_ENV === 'development') {
  // @ts-ignore
  view.submenu.unshift({
    type: 'separator'
  })
  // @ts-ignore
  view.submenu.unshift({
    // @ts-ignore
    label: 'Toggle Developer Tools',
    accelerator: process.platform === 'darwin' ? 'Alt+Command+I' : 'Ctrl+Shift+I',
    click (item: any, focusedWindow: any) {
      if (focusedWindow) focusedWindow.webContents.toggleDevTools()
    }
  })
  // @ts-ignore
  view.submenu.unshift({
    // @ts-ignore
    label: 'Reload',
    accelerator: 'CmdOrCtrl+R',
    click (item: any, focusedWindow: any) {
      if (focusedWindow) focusedWindow.reload()
    }
  })
}

let window = {
  role: 'window',
  submenu: [
    {
      role: 'minimize'
    },
    {
      role: 'close'
    }
  ]
}

let help = {
  role: 'help',
  submenu: [
    {
      label: 'Learn how to create a new wallet'
    },
    {
      label: 'Learn how to restore an existing wallet'
    },
    {
      label: 'Learn how to send XVG'
    },
    {
      label: 'Learn how to receive XVG'
    },
    {
      type: 'separator'
    },
    {
      label: 'Visit our GitHub',
      click () {
        require('electron').shell.openExternal('https://github.com/vergecurrency')
      }
    },
    {
      label: 'Visit vergecurrency.com',
      click () {
        require('electron').shell.openExternal('https://vergecurrency.com')
      }
    }
  ]
}

// @ts-ignore
template.push(actions)
// @ts-ignore
template.push(view)
// @ts-ignore
template.push(window)
// @ts-ignore
template.push(help)

if (process.platform === 'darwin') {
  const name = app.name
  template.unshift({
    label: name,
    // @ts-ignore
    submenu: [
      {
        role: 'about'
      },
      {
        label: 'Check for Updates...'
      },
      {
        type: 'separator'
      },
      {
        label: 'Settings',
        accelerator: 'CmdOrCtrl+,'
      },
      {
        role: 'services',
        submenu: []
      },
      {
        type: 'separator'
      },
      {
        role: 'hide'
      },
      {
        role: 'hideothers'
      },
      {
        role: 'unhide'
      },
      {
        type: 'separator'
      },
      {
        label: 'Quit ' + app.name,
        accelerator: 'CmdOrCtrl+Q',
        click () {
          menuValues.forceQuit = true
          app.quit();
        }
      }
    ]
  })

  // Window menu.
  // @ts-ignore
  template[3].submenu = [
    {
      label: 'Minimize',
      accelerator: 'CmdOrCtrl+M',
      role: 'minimize'
    },
    {
      label: 'Zoom',
      role: 'zoom'
    },
    {
      type: 'separator'
    },
    {
      label: 'Bring All to Front',
      role: 'front'
    }
  ]
}
