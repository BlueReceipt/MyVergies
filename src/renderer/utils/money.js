export const DECIMAL_PER_SATOSHI = 6

export const getFormattedCurrency = (amount, lang) =>
  new Intl.NumberFormat(lang, {
    style: 'currency',
    currency: lang.includes('de') ? 'EUR' : 'USD'
  }).format(amount)

export const getFromattedCrypto = (amount, lang, coin) =>
  new Intl.NumberFormat(lang, {
    style: 'currency',
    currency: coin
  }).format(amount)

export const formatAmountFromSatoshis = (satoshis, lang) => {
  return getFromattedCrypto(satoshis / 10 ** DECIMAL_PER_SATOSHI, lang, 'XVG')
}
