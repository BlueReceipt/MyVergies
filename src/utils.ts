export const isDevelopmentEnvironment = () =>
  process && process.env.NODE_ENV === "development";

export const isProductionEnvironment = () =>
  process && process.env.NODE_ENV === 'production';

export const isMacOSEnvironment = () =>
  process && process.platform === "darwin";

export const isWinOSEnvironment = () =>
  process && process.platform === 'win32';
