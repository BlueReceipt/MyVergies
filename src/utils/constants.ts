export const eventConstants = {
  toggleTor: 'TOGGLE_TOR',
  toggledTor: 'TOR_TOGGLED'
}

export default {
  priceApi: 'https://api.vergecurrency.network/price/api/v1/price',
  ipApi: 'https://api.vergecurrency.network/price/api/v1/ip/',
  vwsApi: 'https://api.vergecurrency.network/vws/api',
  bnApi: 'https://api.vergecurrency.network/node/api/XVG',
  explorer: 'https://verge-blockchain.info',
  termsOfUse: 'https://vergecurrency.com/wallets/terms',
  satoshiDivider: 1000000,
  decimalPerSatoshi: 6,
  defaultCurrencyCode: 'USD',
  paperKeyLength: 12
}
