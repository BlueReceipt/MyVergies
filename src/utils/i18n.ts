import ar from '../assets/i18n/ar'
import bn from '../assets/i18n/bn'
import de from '../assets/i18n/de'
import en from '../assets/i18n/en'
import es from '../assets/i18n/es'
import fr from '../assets/i18n/fr'
import hi from '../assets/i18n/hi'
import it from '../assets/i18n/it'
import ja from '../assets/i18n/ja'
import nl from '../assets/i18n/nl'
import pt from '../assets/i18n/pt'
import ru from '../assets/i18n/ru'
import zh from '../assets/i18n/zh'

const messages = {
  ar,
  bn,
  de,
  en,
  es,
  fr,
  hi,
  it,
  ja,
  nl,
  pt,
  ru,
  zh
}
export default messages
