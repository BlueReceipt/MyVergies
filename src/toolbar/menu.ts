import { MenuItem } from 'electron'
import HelpMenu from './help'
import WindowMenu from './window'
import ViewMenu from './view'
import ServicesMenu from './services'
import AppMenu from './app'
import { isMacOSEnvironment } from '@/utils'

/**
 * macOS Dock menu items
 */
export const dockTemplate = [
  {
    label: 'Send',
    click () {
      console.log('Send')
    }
  },
  {
    label: 'Receive',
    click () {
      console.log('Receive')
    }
  },
  {
    label: 'Settings',
    click () {
      console.log('Settings')
    }
  }
]

/**
 * Application menu.
 */
export const generateMenuTemplate: () => MenuItem[] = () => [
  ...(isMacOSEnvironment() ? [AppMenu] : []),
  ServicesMenu,
  ViewMenu,
  WindowMenu,
  HelpMenu
]
