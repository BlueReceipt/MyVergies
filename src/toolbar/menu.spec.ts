import { generateMenuTemplate } from "./menu"
import utils from '@/utils'

jest.mock('@/utils');
const mockedUtils = utils as jest.Mocked<typeof utils>

test("Menus should always have the same order", () => {
  mockedUtils.isMacOSEnvironment.mockImplementation(() => true)
  expect(generateMenuTemplate()).toHaveLength(5)
  expect(generateMenuTemplate()).toMatchSnapshot()
})

test("Menu should take MacOS into special consideration", () => {
  mockedUtils.isMacOSEnvironment.mockImplementation(() => false)
  expect(generateMenuTemplate()).toHaveLength(4)

  mockedUtils.isMacOSEnvironment.mockImplementation(() => true)
  expect(generateMenuTemplate()).toHaveLength(5)
})
