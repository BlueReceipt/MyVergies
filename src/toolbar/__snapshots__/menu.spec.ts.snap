// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menus should always have the same order 1`] = `
Array [
  Object {
    "label": "MyVergies",
    "role": "appMenu",
    "submenu": Object {
      "append": [MockFunction] {
        "calls": Array [
          Array [
            mockConstructor {
              "values": Object {
                "role": "about",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "label": "Check for Updates...",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "type": "separator",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "accelerator": "CmdOrCtrl+,",
                "click": [Function],
                "label": "Settings",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "role": "services",
                "submenu": Array [],
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "type": "separator",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "role": "hide",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "role": "hideOthers",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "role": "unhide",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "type": "separator",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "accelerator": "CmdOrCtrl+Q",
                "label": "Quit MyVergies",
                "role": "quit",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
  },
  mockConstructor {
    "values": Object {
      "role": "editMenu",
      "submenu": Object {
        "append": [MockFunction] {
          "calls": Array [
            Array [
              mockConstructor {
                "values": Object {
                  "label": "Lock wallet",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "type": "separator",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "click": [Function],
                  "label": "Send",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "click": [Function],
                  "label": "Receive",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "type": "separator",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "accelerator": "CmdOrCtrl+Z",
                  "role": "undo",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "accelerator": "Shift+CmdOrCtrl+Z",
                  "role": "redo",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "type": "separator",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "accelerator": "CmdOrCtrl+X",
                  "role": "cut",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "accelerator": "CmdOrCtrl+C",
                  "role": "copy",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "accelerator": "CmdOrCtrl+V",
                  "role": "paste",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "accelerator": "CmdOrCtrl+A",
                  "role": "selectAll",
                },
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
    },
  },
  mockConstructor {
    "values": Object {
      "role": "viewMenu",
      "submenu": Object {
        "append": [MockFunction] {
          "calls": Array [
            Array [
              mockConstructor {
                "values": Object {
                  "role": "resetZoom",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "role": "zoomIn",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "role": "zoomOut",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "type": "separator",
                },
              },
            ],
            Array [
              mockConstructor {
                "values": Object {
                  "role": "togglefullscreen",
                },
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
    },
  },
  Object {
    "role": "windowMenu",
    "submenu": Object {
      "append": [MockFunction] {
        "calls": Array [
          Array [
            mockConstructor {
              "values": Object {
                "role": "minimize",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "role": "close",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
  },
  Object {
    "role": "help",
    "submenu": Object {
      "append": [MockFunction] {
        "calls": Array [
          Array [
            mockConstructor {
              "values": Object {
                "label": "Learn how to create a new wallet",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "label": "Learn how to restore an existing wallet",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "label": "Learn how to send XVG",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "label": "Learn how to receive XVG",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "type": "separator",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "click": [Function],
                "label": "Visit our GitHub",
              },
            },
          ],
          Array [
            mockConstructor {
              "values": Object {
                "click": [Function],
                "label": "Visit vergecurrency.com",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
  },
]
`;
